neat_ml/tests/test_utils_module.py

```diff
@@ -0,0 +1,180 @@
+from pathlib import Path
+from importlib import resources
+from typing import Callable, Generator, Any, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import numpy.testing as npt
+import pandas as pd
+import pytest
+from matplotlib.testing.compare import compare_images
+from sklearn.mixture import GaussianMixture
+
+from neat_ml.utils import figure_utils
+
+@pytest.fixture(scope="session")
+def synthetic_df() -> pd.DataFrame:
+    """Deterministic (seeded) composition/phase dataframe."""
+    rng = np.random.default_rng(7)
+    x = rng.uniform(0.0, 20.0, 30)
+    y = rng.uniform(0.0, 20.0, 30)
+    phase = (x + y > 20.0).astype(int)
+    return pd.DataFrame({"Sodium Citrate (wt%)": x,
+                         "PEO 8 kg/mol (wt%)": y,
+                         "Phase": phase})
+
+@pytest.fixture(scope="session")
+def baseline_dir() -> Generator[Any, Any, Any]:
+    """
+    Directory that stores the reference (expected) images.
+    """
+    ref = resources.files("neat_ml.tests") / "baseline"
+    with resources.as_file(ref) as path:
+        yield path
+
+def test_standardise_labels_mapping():
+    raw = np.array([9, 9, 1])
+    x_comp = np.array([[10, 10], [11, 11], [0.1, 0.1]])
+    actual_std, mapping = figure_utils._standardise_labels(raw, x_comp)
+    
+    assert mapping[1] == 1
+    assert mapping[9] == 0
+    
+    desired_std = np.array([0, 0, 1])
+    npt.assert_array_equal(actual_std, desired_std)
+
+@pytest.mark.parametrize("z, exp_out",
+    [
+        (
+            np.array([[0, 1],
+                      [1, 0]]),
+            np.array([[1., 0.5],
+                      [0.5, 0.0]]),
+        ),
+        (
+            np.array([[0, 0],
+                      [0, 0]]),
+            None,
+        ),
+    ]
+)
+def test_extract_boundary_from_contour(z, exp_out):
+    xs = ys = np.arange(2)
+    boundary = figure_utils.extract_boundary_from_contour(z, xs, ys, level=0.5)
+    
+    if exp_out is None:
+        assert boundary is exp_out
+    else:
+        npt.assert_array_equal(boundary, exp_out)
```

Comment:
for floats, `assert_allclose` is a bit safer, though it probably doesn't matter too much here